Index: List/ListDemo.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- List/ListDemo.py	(date 1588352319851)
+++ List/ListDemo.py	(date 1588352319851)
@@ -0,0 +1,27 @@
+list1 = ['physics', 'chemistry', 1997, 2000]
+list2 = [1, 2, 3, 4, 5 ]
+list3 = ["a", "b", "c", "d"]
+
+# access element
+print(list1[0])
+print(list3[3])
+print(list1[:]) # ['physics', 'chemistry', 1997, 2000]
+print(list1[1:5]) # ['chemistry', 1997, 2000]
+print(list1[1:]) # ['chemistry', 1997, 2000]
+print(list1[:2]) # ['physics', 'chemistry']
+print(list1[:-1]) # except last element ['physics', 'chemistry', 1997]
+print(list1[-1:])# last element [2000]
+
+# updation
+list1[0] = 'xyz'
+for i in list1:
+    print(i , end = " ")
+print("\n")
+print(list1[:])
+
+
+# Deletion
+print(list1[:], end = " ") #['xyz', 'chemistry', 1997, 2000]
+print("\n")
+del list1[0] # delete element at index 0
+print(list1[:], end = " ") #['chemistry', 1997, 2000]
Index: Dictionary/DictionaryDemo.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- Dictionary/DictionaryDemo.py	(date 1588354900462)
+++ Dictionary/DictionaryDemo.py	(date 1588354900462)
@@ -0,0 +1,6 @@
+dict1 = {1:"test", 2:10}
+print(dict1) #{1: 'test', 2: 10}
+print(dict1[2]) #10
+#print(dict[3]) #TypeError: 'type' object is not subscriptable
+
+print(dict1.key())
\ No newline at end of file
Index: Array/ArrayDemo.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- Array/ArrayDemo.py	(date 1588350556405)
+++ Array/ArrayDemo.py	(date 1588350556405)
@@ -0,0 +1,27 @@
+from array import *
+myArr = array('i',[10,20,30])
+for i in range(len(myArr)):
+   print(myArr[i], end =" ")
+print("\n")
+# insertion : insert at the index
+myArr.insert(1,90)
+for i in myArr:
+    print(i,end = " ")
+print("\n")
+# insertion : insert at the end
+myArr.append(100)
+for i in myArr:
+    print(i, end = " ")
+print("\n")
+# deletion
+myArr.remove(10)
+print("\n")
+for i in myArr:
+    print(i,end = " ")
+print("\n")
+# search
+print(myArr.index(100))
+# Updation
+myArr[0] = 101
+for i in myArr:
+    print(i, end = " ")
Index: Tuples/TupleDemo.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- Tuples/TupleDemo.py	(date 1588353801887)
+++ Tuples/TupleDemo.py	(date 1588353801887)
@@ -0,0 +1,19 @@
+tup1 = ('physics', 'chemistry', 1997, 2000);
+tup2 = (1, 2, 3, 4, 5, 6, 7 );
+
+# Access
+print(tup1[:]) # ('physics', 'chemistry', 1997, 2000)
+print(tup1[:-1]) #('physics', 'chemistry', 1997)
+
+# Updation
+tup3 = (12, 34.56);
+tup4 = ('abc', 'xyz');
+#tup4[0] = 'test' # TypeError: 'tuple' object does not support item assignment
+tup5 = tup3+tup4
+print(tup5) #(12, 34.56, 'abc', 'xyz')
+
+# Deletion
+tup = ('physics', 'chemistry', 1997, 2000);
+print(tup)
+del tup
+print(tup) #NameError: name 'tup' is not defined
\ No newline at end of file
Index: Tree/TreeDemo.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- Tree/TreeDemo.py	(date 1588348642276)
+++ Tree/TreeDemo.py	(date 1588348642276)
@@ -0,0 +1,2 @@
+class Node():
+    de
\ No newline at end of file
diff --git Tuples/__init__.py Tuples/__init__.py
new file mode 100644
diff --git List/__init__.py List/__init__.py
new file mode 100644
diff --git Dictionary/__init__.py Dictionary/__init__.py
new file mode 100644
diff --git Array/__init__.py Array/__init__.py
new file mode 100644
